rsconnect/metadata.py

```diff
@@ -128,7 +128,7 @@ def _get_by_key(self, key: str, default: T | None = None) -> T | None:
         """
         return self._data.get(key, default)
 
-    def _get_by_value_attr(self, attr: str, value: str) -> T | None:
+    def _get_by_value_attr(self, attr: str, value: T) -> T | None:
```

Comment:
`value` here seems like it can't be a `T`, but instead should be a new type variable.

`T` is usually something like `AppMetadata` which is a `TypedDict`. `self._data` is a mapping from `str` to `T`.

`value` has to be the type of `attr` within a `T`; we're selecting a matching `T` based on the value of a specific attribute.